examples/weather/shared/src/weather/events.rs

```diff
@@ -0,0 +1,465 @@
+    FetchFavorite(f64, f64),
+    #[serde(skip)]
+    FetchFavorites,
+    #[serde(skip)]
+    SetWeather(Box<crux_http::Result<crux_http::Response<CurrentResponse>>>),
+    #[serde(skip)]
+    SetFavoriteWeather(
+        Box<crux_http::Result<crux_http::Response<CurrentResponse>>>,
+        f64,
+        f64,
+    ),
+}
+
+pub fn update(event: WeatherEvent, model: &mut Model) -> Command<Effect, Event> {
+    match event {
+        WeatherEvent::Show => Command::event(Event::Favorites(Box::new(FavoritesEvent::Restore)))
+            .then(
+                is_location_enabled().then_send(|result| {
+                    Event::Home(Box::new(WeatherEvent::LocationEnabled(result)))
+                }),
+            ),
+        WeatherEvent::LocationEnabled(enabled) => {
+            model.location_enabled = enabled;
+            if enabled {
+                get_location().then_send(|result| {
+                    Event::Home(Box::new(WeatherEvent::LocationFetched(result)))
+                })
+            } else {
+                Command::done()
+            }
+        }
+        WeatherEvent::LocationFetched(location) => {
+            model.last_location.clone_from(&location);
+            if let Some(loc) = location {
+                update(WeatherEvent::Fetch(loc.lat, loc.lon), model)
+            } else {
+                Command::done()
+            }
+        }
+
+        // Internal events related to fetching weather data
+        WeatherEvent::Fetch(lat, long) => Http::get(WEATHER_URL)
+            .expect_json()
+            .query(&CurrentQueryString {
+                lat: lat.to_string(),
+                lon: long.to_string(),
+                units: "metric",
+                appid: API_KEY.clone(),
+            })
+            .expect("could not serialize query string")
+            .build()
+            .then_send(|result| Event::Home(Box::new(WeatherEvent::SetWeather(Box::new(result))))),
+        WeatherEvent::SetWeather(result) => {
+            let cmd = match *result {
+                Ok(mut response) => {
+                    model.weather_data = response.take_body().unwrap();
+                    render()
+                }
+                Err(_) => render(),
+            };
+
+            // If we have favorites, trigger FetchFavorites after setting weather
+            if model.favorites.is_empty() {
+                cmd
+            } else {
+                cmd.and(Command::event(Event::Home(Box::new(
+                    WeatherEvent::FetchFavorites,
+                ))))
+            }
+        }
+        WeatherEvent::FetchFavorites => {
+            // Create a sequence of commands to fetch weather for each favorite
+            let mut cmd = Command::done();
+
+            for favorite in &model.favorites {
+                cmd = cmd.and(Command::event(Event::Home(Box::new(
+                    WeatherEvent::FetchFavorite(favorite.geo.lat, favorite.geo.lon),
+                ))));
+            }
```

Comment:
Nit - this could be written as a `.map` - `Command` supports collecting from iterator of commands as well.

So something like

```
model.favourites.map(|f| Command::event(...)).collect()
```

Reply:
Another good reason to have a weather API client too - to avoid the indirection through the `FetchFavourite` event